src/Bitmap/BitmapFile.cpp

```diff
@@ -22,6 +21,18 @@ BitmapFile BitmapFile::CreateDefaultIndexed(uint16_t bitCount, uint32_t width, u
 	return bitmapFile;
 }
 
+BitmapFile BitmapFile::CreateDefaultIndexed(uint16_t bitCount, uint32_t width, uint32_t height, const std::vector<Color>& palette)
```

Comment:
Since you're copying the palette anyway, you could potentially pass by-value, and then `std::move` from the parameter to the field. That would effectively move the copy up-front to before the call for lvalue semantics, and pass in a temporary directly for rvalue semantics. That means you can avoid the copy altogether for rvalue semantics, and have only a single constructed `std::vector`.

The example uses in the unit test code would make use of the rvalue semantics.